test/lib/pipeline.test.js

```diff
@@ -299,6 +305,32 @@ describe('Pipeline Model', () => {
         });
     });
 
+    describe('get secrets', () => {
+        it('has a secrets getter', () => {
+            const listConfig = {
+                params: {
+                    pipelineId: pipeline.id
+                },
+                paginate: {
+                    count: 25, // This limit is set by the matrix restriction
+                    page: 1
+                }
+            };
+
+            secretFactoryMock.list.resolves(null);
+            // when we fetch secrets it resolves to a promise
+            assert.isFunction(pipeline.secrets.then);
+            // and a factory is called to create that promise
+            assert.calledWith(secretFactoryMock.list, listConfig);
+
+            // When we call pipeline.secrets again it is still a promise
+            assert.isFunction(pipeline.secrets.then);
+            // ...but the factory was not recreated, since the promise is stored
+            // as the model's pipeline property, now
```

Comment:
It's written in a non-idiomatic manner; I had to look at it multiple times to understand what it was trying to accomplish. 

My initial comment is an educational comment and not a functional one, especially since it's a test. Tests are low priority and have been treated as such. They'll be copy and pasted throughout the next PRs too. A PR to fix all the tests to be consistent isn't valuable.